### PR TITLE
Db 1754 mk bug fix context router

### PR DESCRIPTION
**Pull request by kimminahm**
Fixed FABS visual sync. Removed the context.history console warning about deprecated code. Made sure behaviour from validation, back to data entry, back to validation, and any other combination would continue to work

### CODE REVIEW

File: src/js/components/uploadDetachedFiles/UploadDetachedFilesPage.jsx [hunk 1]
```diff
@@ -27,24 +29,47 @@ export default class UploadDetachedFilesPage extends React.Component {
 	}
 
 	componentDidMount(){
+		let showMeta = true;
 		if(this.props.params.submissionID){
 			this.props.setSubmissionId(this.props.params.submissionID);
+			showMeta = false;
+		}else{
```
Comment (alburde1) [inline src/js/components/uploadDetachedFiles/UploadDetachedFilesPage.jsx]: This else isn't necessary, it's already true

File: src/js/components/uploadDetachedFiles/UploadDetachedFilesPage.jsx [hunk 2]
```diff
@@ -16,6 +16,8 @@ import UploadDetachedFileValidation from './UploadDetachedFileValidation.jsx';
```
Comment (alburde1) [inline src/js/components/uploadDetachedFiles/UploadDetachedFilesPage.jsx]: Overall question for this file, why are we even bothering with a `showMeta` state? The only time it comes into play other than for setting itself is deciding which to display. Can't you just base it off the params? Or does it not re-render if that's the case?

File: src/js/components/uploadDetachedFiles/UploadDetachedFilesPage.jsx [hunk 3]
```diff
@@ -27,24 +29,47 @@ export default class UploadDetachedFilesPage extends React.Component {
 	}
 
 	componentDidMount(){
+		let showMeta = true;
 		if(this.props.params.submissionID){
```
Comment (alburde1) [inline src/js/components/uploadDetachedFiles/UploadDetachedFilesPage.jsx]: Do we need this part here anymore if we have `componentWillReceiveProps`?

File: src/js/components/uploadDetachedFiles/UploadDetachedFileMeta.jsx [hunk 4]
```diff
@@ -144,8 +144,9 @@ export default class UploadDetachedFileMeta extends React.Component {
                     // TODO: Remove this when this is eventually tied to user accounts
                     this.props.setSubmissionId(submissionID);
 					this.checkFileStatus(submissionID);
-					this.props.history.push('/uploaddetachedfiles/'+submissionID);
-					this.props.validate();
+					// this.props.history.push('/uploaddetachedfiles/'+submissionID);
+					// this.context.router.transitionTo('/uploadDetachedfiles/'+submissionID);
```
Comment (kimminahm) [inline src/js/components/uploadDetachedFiles/UploadDetachedFileMeta.jsx]: done

File: src/js/components/uploadDetachedFiles/UploadDetachedFileMeta.jsx [hunk 5]
```diff
@@ -157,6 +158,9 @@ export default class UploadDetachedFileMeta extends React.Component {
 
 	checkFileStatus(submissionID) {
 		// callback to check file status
+		if(!submissionID){
```
Comment (kimminahm) [inline src/js/components/uploadDetachedFiles/UploadDetachedFileMeta.jsx]: Sorry, got desperate to fix errors and at one point this was receiving submissionID = null. No longer seems to be the case

File: src/js/components/uploadDetachedFiles/UploadDetachedFileValidation.jsx [hunk 6]
```diff
@@ -53,20 +54,30 @@ export default class UploadDetachedFileValidation extends React.Component {
...
+		}
+	}
+
 	componentWillUnmount() {
 		this.isUnmounted = true;
 	}
 
 	checkFileStatus(submissionID) {
 		// callback to check file status
+		if(!submissionID){
```
Comment (kimminahm) [inline src/js/components/uploadDetachedFiles/UploadDetachedFileValidation.jsx]: same as above

File: src/js/components/uploadDetachedFiles/UploadDetachedFilesPage.jsx (hunk 2 above)
Comment (kimminahm) [inline src/js/components/uploadDetachedFiles/UploadDetachedFilesPage.jsx]: State changes fire off the page to re-render. using only param changes may result in the page refusing to load.

File: src/js/components/uploadDetachedFiles/UploadDetachedFilesPage.jsx (hunk 3 above)
Comment (kimminahm) [inline src/js/components/uploadDetachedFiles/UploadDetachedFilesPage.jsx]: ComponentWillReceiveProps is only fired off when the props change. ComponentDidMount is necessary for the first setup.

File: src/js/components/uploadDetachedFiles/UploadDetachedFilesPage.jsx (hunk 1 above)
Comment (kimminahm) [inline src/js/components/uploadDetachedFiles/UploadDetachedFilesPage.jsx]: good catch

File: src/js/components/uploadDetachedFiles/UploadDetachedFilesPage.jsx (hunk 2 above)
Comment (kimminahm) [inline src/js/components/uploadDetachedFiles/UploadDetachedFilesPage.jsx]: I tried removing all the showMeta stuff and replaced with checking for the props and it didn't work

**Pull request merged**